omnigibson/envs/env_base.py

```diff
@@ -80,6 +92,21 @@ def __init__(
         # Load this environment
         self.load()
 
+        # Set up external cameras
+        self._external_cameras = {}
+        for ec_kwargs in external_camera_kwargs:
+            camera = VisionSensor(**ec_kwargs)
+            camera.load()
+            camera.initialize()
+
+            camera_name = ec_kwargs["name"]
+            if camera_name in external_camera_poses:
```

Comment:
maybe let's force a camera pose to be specified per camera, and assume the number of cam kwargs specified correspond to the number poses specified too.

So then we could do `for ec_kwargs, ec_pose in zip(external_camera_kwargs, external_camera_poses):`